### PR TITLE
Simplify writer usage, add basic unit tests

### PR DESCRIPTION
**Pull request by jcelerier**

### CODE REVIEW

File: tests/unit/midifile_write.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+TEST_CASE("write a track with a note", "[midi_writer]" )
+{
+  libremidi::writer writer;
+  writer.tracks.push_back(
+    libremidi::midi_track{
+      libremidi::track_event{0, 0, libremidi::message::note_on(1, 45, 35)}});
+  std::ofstream empty;
+  writer.write(empty);
+}
```
Comment (keryell) [inline tests/unit/midifile_write.cpp:46]: Missing ROL

Comment (jcelerier) [reply tests/unit/midifile_write.cpp:46]: what do you mean ?

Comment (keryell) [reply tests/unit/midifile_write.cpp:46]: That your file misses an end-of-line character. This is displayed by my GitHub config (perhaps with "Refined GitHub" extension).
No crazy design issue here... :-)

Comment (jcelerier) [reply tests/unit/midifile_write.cpp:46]: oh I see. This is undefined behaviour in C but fine in C++. I can confirm that it comes from the extension, I'll open them an issue :p

**Pull request merged**